host-controller/src/main/java/org/jboss/as/domain/controller/operations/DomainDeploymentOverlayRedeployLinksHandler.java

```diff
@@ -0,0 +1,84 @@
+import org.jboss.as.controller.descriptions.common.ControllerResolver;
+import org.jboss.as.domain.controller.logging.DomainControllerLogger;
+import org.jboss.as.server.deploymentoverlay.AffectedDeploymentOverlay;
+import org.jboss.dmr.ModelNode;
+
+/**
+ * Handler to redeploy deployments linked to an overlay in domain mode.
+ * @author Emmanuel Hugonnet (c) 2017 Red Hat, inc.
+ */
+public class DomainDeploymentOverlayRedeployLinksHandler implements OperationStepHandler {
+
+    public static final StringListAttributeDefinition RUNTIME_NAMES_DEFINITION =
+            new StringListAttributeDefinition.Builder(DEPLOYMENTS)
+                            .setRequired(false)
+                            .build();
+
+    public static final OperationDefinition REDEPLOY_LINKS_DEFINITION = new SimpleOperationDefinitionBuilder(
+            REDEPLOY_LINKS, ControllerResolver.getResolver(ModelDescriptionConstants.DEPLOYMENT_OVERLAY))
+            .addParameter(RUNTIME_NAMES_DEFINITION)
+            .build();
+
+    private final boolean domainRoot;
+
+    public DomainDeploymentOverlayRedeployLinksHandler(boolean domainRoot) {
+        this.domainRoot = domainRoot;
+    }
+
+    @Override
+    public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+        final Set<String> names = domainRoot ? AffectedDeploymentOverlay.listAllLinks(context, context.getCurrentAddressValue()) : AffectedDeploymentOverlay.listLinks(context, context.getCurrentAddress());
+        Set<String> runtimeNames = domainRoot ? AffectedDeploymentOverlay.listAllLinks(context, context.getCurrentAddressValue()) : AffectedDeploymentOverlay.listLinks(context, context.getCurrentAddress());
+        if(operation.hasDefined(RUNTIME_NAMES_DEFINITION.getName())) {
+            Set<String> requiredRuntimeNames = new HashSet<>(RUNTIME_NAMES_DEFINITION.unwrap(context, operation));
+            if(!AffectedDeploymentOverlay.checkRequiredNames(requiredRuntimeNames, runtimeNames).isEmpty() && !isRedeployAfterRemoval(operation)) {
+                throw DomainControllerLogger.ROOT_LOGGER.redeployingUnaffactedRuntimes();
+            }
+            runtimeNames = requiredRuntimeNames;
+        }
+        ModelNode removeOperation = null;
+        if(isRedeployAfterRemoval(operation)) {
+            removeOperation = Operations.createRemoveOperation(context.getCurrentAddress().append(DEPLOYMENT, runtimeNames.iterator().next()).toModelNode());
+        }
+        if (domainRoot) {
+            AffectedDeploymentOverlay.redeployLinksAndTransformOperationForDomain(context, runtimeNames, removeOperation);
+        } else {
+            AffectedDeploymentOverlay.redeployLinksAndTransformOperation(context, removeOperation, context.getCurrentAddress().getParent(), runtimeNames);
+        }
+    }
+
+    private boolean isRedeployAfterRemoval(ModelNode operation) {
+        return operation.hasDefined(REMOVE) && operation.get(REMOVE).asBoolean();
```

Comment:
I set it in https://github.com/wildfly/wildfly-core/pull/2115/files/112f59a8dda4d52f86fd32a1f3635d6c6cb6a462#diff-9e579e339e84bbfe7b4638d2f6e40394R68
This is not a 'real' parameter. Maybe I should rename it